code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -386,3 +389,17 @@
 	GLOB.offered_mob_list -= src
 	AddComponent(/datum/component/ai_controller, /datum/ai_behavior/xeno)
 	a_intent = INTENT_HARM
+
+/// Handles logic for weeds nearby the xeno getting removed
+/mob/living/carbon/xenomorph/proc/handle_weeds_adjacent_removed(datum/source)
+	SIGNAL_HANDLER
+	var/obj/effect/alien/weeds/found_weed = locate(/obj/effect/alien/weeds) in loc
+	if(!QDELING(found_weed))
+		return
```

Comment:
Why this check here and why isnt it qdeleted

Reply:
From what I understand, the `COMSIG_LIVING_WEEDS_ADJACENT_REMOVED` signal is sent before the thing is deleted. Although, now that I checked again, hivemind code uses "QDESTROYING" instead so that may be better.

Reply:
Replaced `QDELING `with `QDESTROYING`